.github/workflows/issue-opened.yml

```
@@ -10,61 +15,86 @@ permissions:
   id-token: write
 
 jobs:
+  config:
+    runs-on: "ubuntu-latest"
+    outputs:
+      has-secrets: ${{ steps.check.outputs.has-secrets }}
+    steps:
+      - name: "Check for secrets"
+        id: check
+        shell: bash
+        run: |
+          if [ "${{ github.repository }}" == "grafana/grafana" ]; then
+            echo "has-secrets=1" >> "$GITHUB_OUTPUT"
+          fi
+
   main:
     runs-on: ubuntu-latest
+    needs: config
+    if: needs.config.outputs.has-secrets
```

```suggestion
    if: github.repository == 'grafana/grafana'
```

I don't see any reason to spawn a "config" job at all if you can just do this check directly. Just add this check and remove the whole `config` job above.

---

Good idea!